ghzdac/correction.py

```diff
@@ -1165,10 +1188,17 @@ def DACifyFT(self, signal, t0=0, n=8192, offset=0, nfft=None, loop=False,
                 #decay times:
                 # add to qubit registry the following keys:
                 # settlingAmplitudes=[-0.05]  #relative amplitude
-                # settlingRates = [0.012]    #rate is in GHz, (1/ns)
+                # settlingRates = [0.01]    #rate is in GHz, (1/ns)
                 if np.alen(decayRates):
-                    freqs = 2j*np.pi*freqs
-                    precalc /= (1.0 + np.sum(decayAmplitudes[:, None] * freqs[None, :] / (freqs[None, :] + decayRates[:, None]), axis=0))
+                    precalc /= (1.0 + np.sum(decayAmplitudes[:, None] * 2j*np.pi*freqs[None, :] / (2j*np.pi*freqs[None, :] + decayRates[:, None]), axis=0))
+
+                #reflections:
+                # add to qubit registry the following keys:
+                # reflectionAmplitudes=[0.05]  #relative amplitude
+                # reflectionRates = [0.01]    #rate is in GHz, (1/ns)
+                if np.alen(reflectionRates):
+                    for rate,amplitude in zip(reflectionRates,reflectionAmplitudes):
+                        precalc /= (1.0 - amplitude) / (1.0-amplitude*np.exp(-2j*np.pi*freqs/rate))
```

Comment:
It makes sense to me that in general you want to convolve the response functions, but I guess it depends on how you measure the reflection response. If you measure two reflections peaks in the time domain, say, then you get a time-domain response like `X(t) + a*X(t-t_a) + b*X(t-t_b)` which fourier transforms to something like `x(f) * (1 + a*e(-2pi f t_a) + b*e(-2pi f t_b))`, so now to correct for this you want to divide by the sum of the two reflection terms instead of the product. Obviously, I'm playing fast and loose with signs here, but I guess what I'd like to see here is some more detail of how were measuring and modeling the reflection response, so that we can be sure that the correction is being applied in a manner consistent with those measurements.